federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/control/PasswordPolicyResponseControl.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.storage.ldap.idm.store.ldap.control;
+
+import java.math.BigInteger;
+
+import javax.naming.ldap.Control;
+
+import org.jboss.logging.Logger;
+import org.wildfly.security.asn1.ASN1;
+import org.wildfly.security.asn1.ASN1Exception;
+import org.wildfly.security.asn1.DERDecoder;
+
+public class PasswordPolicyResponseControl implements Control {
+
+    public static final String OID = "1.3.6.1.4.1.42.2.27.8.5.1";
```

Comment:
How safe it is this control in terms of support from the different LDAP vendors? Do want a switch to allow administrators to decide if they want this new behavior in their LDAP integrations?

Reply:
The control is the only "standard" (draft) I know that targets this use case, but from implementation I know only OpenLDAP and 389ds (as well as SSSD as client side implementation). Lets add a switch.